distribution/lib/Standard/Base/0.0.0-dev/src/Network/Reload_Detector.enso

```diff
@@ -0,0 +1,30 @@
+import Standard.Base.Error.Error
+import Standard.Base.Meta
+import Standard.Base.Nothing.Nothing
+import Standard.Base.Runtime.Managed_Resource.Managed_Resource
+import Standard.Base.Runtime.Ref.Ref
+from Standard.Base.Data.Boolean import Boolean, True, False
+
+## PRIVATE
+   This is used by ReloadDetector.java to create a `Managed_Resource` that is
+   garbage collected when the reload button is pressed.
+
+   The managed resource contains a Ref containing a 0 (the value is
+   unimportant). When the reload button is pressed, the ref is removed and
+   attempting to access it using `with` throws an `Uninitialized_State`. When
+   the `Uninitialized_State` is detected, it indicates that the reload has been
+   initiated.
+type Reload_Detector
+    private Value mr:Managed_Resource
+
+    new -> Reload_Detector =
+        mr = Managed_Resource.register (Ref.new 1) (x-> Nothing) True
+        Reload_Detector.Value mr
+
+    has_reload_occurred self =
+        self.mr.has_been_finalized
+
+    simulate_reload_test_only self =
+        self.mr.finalize
```

Comment:
this should be a static method on module.

Reply:
Done